tasks/eth/006-MCP-L1/postCheck.s.sol

```diff
@@ -0,0 +1,231 @@
+        require(config.batcherHash() == bytes32(uint256(uint160(cfg.batchSenderAddress()))));
+        require(config.gasLimit() == uint64(cfg.l2GenesisBlockGasLimit()));
+        require(config.unsafeBlockSigner() == cfg.p2pSequencerAddress());
+        // Check _config
+        ResourceMetering.ResourceConfig memory rconfig;
+        rconfig.maxResourceLimit = Constants.DEFAULT_RESOURCE_CONFIG().maxResourceLimit;
+        rconfig.elasticityMultiplier = Constants.DEFAULT_RESOURCE_CONFIG().elasticityMultiplier;
+        rconfig.baseFeeMaxChangeDenominator = Constants.DEFAULT_RESOURCE_CONFIG().baseFeeMaxChangeDenominator;
+        rconfig.minimumBaseFee = Constants.DEFAULT_RESOURCE_CONFIG().minimumBaseFee;
+        rconfig.systemTxMaxGas = Constants.DEFAULT_RESOURCE_CONFIG().systemTxMaxGas;
+        rconfig.maximumBaseFee = Constants.DEFAULT_RESOURCE_CONFIG().maximumBaseFee;
+
+        require(resourceConfig.maxResourceLimit == rconfig.maxResourceLimit);
+        require(resourceConfig.elasticityMultiplier == rconfig.elasticityMultiplier);
+        require(resourceConfig.baseFeeMaxChangeDenominator == rconfig.baseFeeMaxChangeDenominator);
+        require(resourceConfig.systemTxMaxGas == rconfig.systemTxMaxGas);
+        require(resourceConfig.minimumBaseFee == rconfig.minimumBaseFee);
+        require(resourceConfig.maximumBaseFee == rconfig.maximumBaseFee);
+        // Depends on start block being set to 0 in `initialize`
+        uint256 cfgStartBlock = cfg.systemConfigStartBlock();
+        require(config.startBlock() == (cfgStartBlock == 0 ? block.number : cfgStartBlock));
+        require(config.batchInbox() == cfg.batchInboxAddress());
+        // Check _addresses
+        require(config.l1CrossDomainMessenger() == prox.L1CrossDomainMessenger);
+        require(config.l1ERC721Bridge() == prox.L1ERC721Bridge);
+        require(config.l1StandardBridge() == prox.L1StandardBridge);
+        require(config.l2OutputOracle() == prox.L2OutputOracle);
+        require(config.optimismPortal() == prox.OptimismPortal);
+        require(config.optimismMintableERC20Factory() == prox.OptimismMintableERC20Factory);
+    }
+
+    /// @notice Asserts that the L1CrossDomainMessenger is setup correctly
+    function checkL1CrossDomainMessenger() internal view {
+        console.log("Running chain assertions on the L1CrossDomainMessenger");
+        L1CrossDomainMessenger messenger = L1CrossDomainMessenger(prox.L1CrossDomainMessenger);
+
+        require(address(messenger.OTHER_MESSENGER()) == Predeploys.L2_CROSS_DOMAIN_MESSENGER);
+        require(address(messenger.otherMessenger()) == Predeploys.L2_CROSS_DOMAIN_MESSENGER);
+
+        require(address(messenger.PORTAL()) == prox.OptimismPortal);
+        require(address(messenger.portal()) == prox.OptimismPortal);
+        require(address(messenger.superchainConfig()) == prox.SuperchainConfig);
+        bytes32 xdmSenderSlot = vm.load(address(messenger), bytes32(uint256(204)));
```

Comment:
Let's either:
1. Easier: Add a link to the snapshots so this can be validated, and pull this assumption/declaration to the top of the file so it's more visible
2. Harder: Read from the snapshots, by either having the monorepo at the expected tag as a dependency, or taking the path as an input

Reply:
implemented approach 1. lmk if this is correct! https://github.com/ethereum-optimism/superchain-ops/blob/8630b8c96e7f6244c33bbf3d8b15176282c61148/tasks/eth/006-MCP-L1/SignFromJsonWithAssertions.s.sol#L22-L24

Reply:
Thank you!